MacOS: Patch or update Bullet library

(Follow up issue of https://github.com/panda3d/panda3d/issues/767) Unless compiling with SDK 10.6, bullet compilation fails with the following errors:

<pre>
panda3d-thirdparty/build/bullet-prefix/src/bullet/src/LinearMath/btVector3.h:330:7: error: argument value 10880 is outside the valid range [0, 255] [-Wargument-outside-range]
panda3d-thirdparty/build/bullet-prefix/src/bullet/src/LinearMath/btMatrix3x3.h:882:17: error: argument value 10880 is outside the valid range [0, 255] [-Wargument-outside-range]
</pre>

This has been fixed upstream, https://github.com/bulletphysics/bullet3/pull/2232 but we need to either update the version of bullet downloaded or patch the code before compiling.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -202,7 +202,7 @@ if(BUILD_BULLET)
   ExternalProject_Add(
     bullet
     GIT_REPOSITORY https://github.com/bulletphysics/bullet3.git
-    GIT_TAG 2fb92bc40c16e4da5a9018479d4a8e1899702ab8 #v2.83.7
+    GIT_TAG 7638b7c5a659dceb4e580ae87d4d60b00847ef94 # https://github.com/bulletphysics/bullet3/pull/2232
     #URL https://github.com/bulletphysics/bullet3/archive/2.83.7.tar.gz
     #URL_HASH SHA256=919087c764d0e93555c6ef0830c8951a9caafea99f52ac8b366e96adbe0112ed
 

